lib/smart_proxy_ansible/runner/ansible_runner.rb

```diff
@@ -83,10 +83,18 @@ def handle_host_event(hostname, event)
       def handle_broadcast_data(event)
         log_event("broadcast", event)
         if event['event'] == 'playbook_on_stats'
+          failures = event.dig('event_data', 'failures') || {}
           header, *rows = event['stdout'].strip.lines.map(&:chomp)
           @outputs.keys.select { |key| key.is_a? String }.each do |host|
             line = rows.find { |row| row =~ /#{host}/ }
             publish_data_for(host, [header, line].join("\n"), 'stdout')
+
+            # If the task has been rescued, it won't consider a failure
+            if @exit_statuses[host] && @exit_statuses[host] != 0
+              if failures[host].nil? || failures[host] <= 0
```

Comment:
Can there be less than zero failures?

Reply:
I think no. It is just my usual habit for safety. I can do '==' if you don't like it.